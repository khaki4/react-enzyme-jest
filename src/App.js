import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';

const Test = () => <div>Testing</div>

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      on: false,
      input: '',
      mainColor: 'blue',
    }
  }

  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <p>
            Edit <code>src/App.js</code> and save to reload.
          </p>
          Test
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Hello World
          </a>
          <h3 className={this.state.mainColor}>Everyone is Welcome!</h3>
        </header>
        <p className="button-state">{this.state.on ?
          'Yes!' : 'No!'
        }</p>
        <button
          onClick={() => this.setState({ on: true })}
        >Click</button>
        <h2>{this.state.input}</h2>
        <input type="text"  onChange={(e) => this.setState({ input: e.currentTarget.value })} />
      </div>
    );
  }
}

export class Link extends Component {
  render() {
    return this.props.hide ? null : <a href={this.props.address}>Click</a>
  }
}

export default App;
