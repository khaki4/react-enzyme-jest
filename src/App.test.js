import React from 'react';
import ReactDOM from 'react-dom';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'; 
import toJson from 'enzyme-to-json';
import App from './App';

configure({ adapter: new Adapter() });

describe('<App />', () => {
  it('should contain 1 element', () => {
    const wraaper = shallow(<App />);
    expect(wraaper.find('p').exists()).toBe(true);
    // expect(wraaper.find('ul').children().length).toBe(3);
    // expect(wraaper.find('ul').hasClass('kevin')).toBe(true);
  });

  it('matches the snapshot', () => {
    const tree = shallow(<App />);
    expect(toJson(tree)).toMatchSnapshot()
  });

});