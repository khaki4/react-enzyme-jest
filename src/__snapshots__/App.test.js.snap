// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    Test
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Hello World
    </a>
  </header>
</div>
`;
